File: app/src/main/java/za/foundation/praekelt/mama/inject/module/ApplicationModule.java
package za.foundation.praekelt.mama.inject.module;

import android.content.Context;

import com.squareup.otto.Bus;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import za.foundation.praekelt.mama.app.App;

/**
 * Created by eduardokolomajr on 2015/09/10.
 */
@Module
public class ApplicationModule {
    private App app;

    public ApplicationModule(App app) {
        this.app = app;
    }

    @Provides
    @Singleton
    public Context provideApplicationContext() {
        return app;
    }

    @Provides
    @Singleton
    public App provideApplication()

    {
        return app;
    }

    @Provides
    @Singleton
    public Bus provideBus() {
        return app.getBus();
    }
}

File: app/src/main/kotlin/za/foundation/praekelt/mama/inject/module/ApplicationModule.kt
package za.foundation.praekelt.mama.inject.module

import android.content.Context
import com.squareup.otto.Bus
import dagger.Module
import dagger.Provides
import za.foundation.praekelt.mama.app.App
import javax.inject.Singleton

/**
 * Dagger Application Module
 */
Module
class ApplicationModule(val app: App){
    Provides
    Singleton
    fun provideApplicationContext(): Context{
        return app
    }

    Provides
    Singleton
    fun provideApplication(): App {
        return app
    }

    Provides
    Singleton
    fun provideBus(): Bus {
        return app.bus
    }
}
